pkg/controller/shoot.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+)
+
+// Shoot provides convenience methods for getting various Shoot properties.
+type Shoot struct {
```

Comment:
Instead of defining another `Shoot` type, please write these functions here as dedicated helper functions taking a regular `Shoot` as argument.

Reply:
Are you sure? I don't see how this helps or adds any clarity. The preferred style in Go (and other languages) for helpers that act on a common types would be to add them as methods. @rfranzke, your opinion?

Reply:
Actually, there is one good reason for not defining a `Shoot` type and it's that you wouldn't need to create an instance in order to work with the shoot, which makes the client code slightly less complex. So reworked as suggested.